accounts/abi/bind/precompiletemplate.go

```diff
@@ -0,0 +1,366 @@
+	// Assumes that [input] does not include selector
+	// You can use unpacked [inputStruct] variable in your code
+	inputStruct, err := Unpack{{capitalise .Normalized.Name}}Input(input)
+	if err != nil{
+		return nil, remainingGas, err
+	}
+	{{- end}}
+
+	{{if not .Original.IsConstant | and $contract.AllowList}}
+	// Allow list is enabled and {{.Normalized.Name}} is a state-changer function.
+	// This part of the code restricts the function to be called only by enabled/admin addresses in the allow list.
+	// You can modify/delete this code if you don't want this function to be restricted by the allow list.
+	stateDB := accessibleState.GetStateDB()
+	// Verify that the caller is in the allow list and therefore has the right to modify it
+	callerStatus := getAllowListStatus(stateDB, {{$contract.Type}}Address, caller)
+	if !callerStatus.IsEnabled() {
+		return nil, remainingGas, fmt.Errorf("%w: %s", ErrCannot{{.Normalized.Name}}, caller)
+	}
+	// allow list code ends here.
+  {{end}}
+	// CUSTOM CODE STARTS HERE
+	// use inputStruct ...
+	{{- if len .Normalized.Outputs | eq 0}}
+	// this function does not return an output, leave this one as is
+	packedOutput := []byte{}
+	{{- else}}
+	packedOutput, err := Pack{{.Normalized.Name}}Output(nil)
+	if err != nil {
+		return nil, remainingGas, err
+	}
+	{{- end}}
+
+	// Return the packed output and the remaining gas
+	return packedOutput, remainingGas, nil
+}
+{{end}}
+
+{{- if .Contract.Fallback}}
+{{- with .Contract.Fallback}}
+// {{decapitalise $contract.Type}}Fallback executed if a function identifier does not match any of the available functions in a smart contract.
+// This function cannot take an input or return an output.
+func {{decapitalise $contract.Type}}Fallback (accessibleState PrecompileAccessibleState, caller common.Address, addr common.Address, _ []byte, suppliedGas uint64, readOnly bool) (_ []byte, remainingGas uint64, err error) {
+	if remainingGas, err = deductGas(suppliedGas, {{$contract.Type}}FallbackGasCost); err != nil {
+		return nil, 0, err
+	}
+
+	if readOnly {
+		return nil, remainingGas, vmerrs.ErrWriteProtection
+	}
+
+	{{- if $contract.AllowList}}
+	// Allow list is enabled and {{.Normalized.Name}} is a state-changer function.
+	// This part of the code restricts the function to be called only by enabled/admin addresses in the allow list.
+	// You can modify/delete this code if you don't want this function to be restricted by the allow list.
+	stateDB := accessibleState.GetStateDB()
+	// Verify that the caller is in the allow list and therefore has the right to modify it
+	callerStatus := getAllowListStatus(stateDB, {{$contract.Type}}Address, caller)
+	if !callerStatus.IsEnabled() {
+		return nil, remainingGas, fmt.Errorf("%w: %s", Err{{$contract.Type}}CannotFallback, caller)
+	}
+	// allow list code ends here.
+	{{- end}}
+
+	// CUSTOM CODE STARTS HERE
+
+	// Fallback function cannot return an output, return remainingGas only.
```

Comment:
A fallback function can return an output: https://docs.soliditylang.org/en/latest/contracts.html?highlight=fallback%20function. It is just the receive function which is not allowed to return an output.